bin/update-news

```diff
@@ -0,0 +1,124 @@
+#! /usr/bin/env python
+"""
+This is a script to help us keep our "blog/news.json" file up to
+date. It will scan the "blog" directory, sort items by date (which
+is just title) and then read the metadata from the entries to update
+the "news.json" file.
+"""
+from __future__ import annotations
+
+import datetime
+import json
+import os
+import pathlib
+import re
+from typing import Sequence, Any, Iterator
+
+import ruamel_yaml as yaml
+
+
+#: Directory where blog posts are located
+BLOGS_DIR = pathlib.Path("./blog")
+
+#: Location of the news file
+NEWS_FILE = BLOGS_DIR.joinpath("news.json")
+
+#: Matches anything that begins with a time stamp (e.g. 2023-03-23-something)
+BLOGS_TITLE_PAT = re.compile(r"([0-9]{4}-[0-9]{2}-[0-9]{2})(.+)")
+
+
+def has_blog_title_pattern(item: str) -> bool:
+    """
+    Filter that checks whether the file or directory has the pattern we want
+    (see the above regex for more information). 
+
+    Example:
+        - 2023-04-02-blog-post.md
+        - 2023-04-04-another-post/
+    """
+    match = BLOGS_TITLE_PAT.match(item)
+
+    return match is not None
+
+def has_index_file_ending(item: pathlib.Path) -> bool:
+    """
+    Checks to see if the file ends with a *.md or *.mdx
+    """
+    return item.name.endswith("index.md") or item.name.endswith("index.mdx")
```

Comment:
```suggestion
    return item.name.lower() in ("index.md", "index.mdx")
```

I'd write it like this, otherwise it will turn positive for titles like `regenerating_the_conda_index.md`.